specs/interop.md

```diff
@@ -0,0 +1,708 @@
+
+This inclusion property prevents a class of attacks where the user can trick the derivation pipeline
+into reorganizing the sequencer.
+
+### Safety
+
+Safety is an abstraction that is useful for reasoning about security. It should be thought about
+as a spectrum from `unsafe` to `final`. Users can choose to operate on information based on its
+level of safety depending on their risk profile and personal preferences.
+
+The following labels are used to describe both inputs and outputs:
+
+- `unsafe`
+- `cross-unsafe`
+- `safe`
+- `finalized`
+
+Inputs correspond to the inputs to the state transition function while outputs correspond to the side
+effects of the state transition function.
+
+Anything before `safe` technically uses a "preconfirmation" based security model which is not part
+of consensus. While useful to have definitions of the default meanings of these terms, they are
+technically policy and may be changed in the future.
+
+The `unsafe` label has the lowest latency while the `finalized` label has the highest latency.
+A set of properties must be matched before an input or an output can be promoted to the next
+label.
+
+The initiating messages for all dependent executing messages MUST be resolved as safe before an L2 block can transition
+from being unsafe to safe. Users MAY optimistically accept unsafe blocks without any verification of the
+executing messages. They SHOULD optimistically verify the initiating messages exist in destination unsafe blocks
+to more quickly reorganize out invalid blocks.
+
+#### `unsafe` Inputs
+
+- MUST be signed by the p2p sequencer key
+- MAY be reorganized
+- MUST be promoted to a higher level of safety or reorganized out to ensure liveness
+
+`unsafe` inputs are currently gossiped around the p2p network. To prevent denial of service, they MUST
+be signed by the sequencer. This signature represents the sequencer's claim that it
+built a block that conforms to the protocol. `unsafe` blocks exist to give low latency access to the
+latest information. To keep the latency as low as possible, cross chain messages are assumed valid
+at this stage. This means that the remote unsafe inputs are trusted.
+
+An alternative approach to `unsafe` inputs would be to include an SGX proof that the sequencer ran
+particular software when building the block.
+
+#### `cross-unsafe` Inputs
+
+- MUST have valid cross chain messages
+
+`cross-unsafe` represents the `unsafe` blocks that had their cross chain messages fully verified.
+The network can be represented as a graph where each block across all chains are represented as
+a node and then a directed edge between two blocks represents the source block of the initiating
+message and the block that included the executing message.
+
+An input can be promoted from `safe` to `cross-unsafe` when the full dependency graph is resolved
+such that all cross chain messages are verified to be valid and at least one message in the dependency
+graph is still `unsafe`.
```

Comment:
This was / is kind of confusing to me when reading it. Why would an input be `safe` is at least one message in the dependency graph is still `unsafe`, since the definition of `safe` requires all messages in the graph to be resolved? Also, wouldn't it be "demoted", not "promoted" from `safe` to `cross-unsafe`?

Reply:
This was a typo, should be updated